src/ui-test/tests/debugger.test.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai';
+import * as path from 'path';
+import {
+    ActivityBar,
+    DebugView,
+    EditorView,
+    SideBarView,
+    TextEditor,
+    VSBrowser,
+    ViewSection,
+    WebDriver
+} from "vscode-uitests-tooling";
+import {
+    CAMEL_ROUTE_YAML_WITH_SPACE,
+    CAMEL_RUN_DEBUG_ACTION_LABEL,
+    HELLO_CAMEL_MESSAGE,
+    HELLO_WORLD_MESSAGE,
+    TEST_ARRAY_RUN_DEBUG,
+    activateTerminalView,
+    disconnectDebugger,
+    executeCommand,
+    killTerminal,
+    waitUntilTerminalHasText
+} from '../utils';
+
+describe('Camel Debugger tests', function () {
+    this.timeout(240000);
+
+    let driver: WebDriver;
+
+    before('Before setup', async function () {
+        driver = VSBrowser.instance.driver;
+
+        await VSBrowser.instance.openResources(path.resolve('src', 'ui-test', 'resources'));
+
+        await (await new ActivityBar().getViewControl('Explorer')).openView();
+
+        const section = await new SideBarView().getContent().getSection('resources') as ViewSection;
+        await section.openItem(CAMEL_ROUTE_YAML_WITH_SPACE);
+
+        const editorView = new EditorView();
+        await driver.wait(async function () {
+            return (await editorView.getOpenEditorTitles()).find(title => title === CAMEL_ROUTE_YAML_WITH_SPACE);
+        }, 5000);
+    });
+
+    after('After cleanup', async function () {
+        await driver.wait(async function () {
+            return !await new TextEditor().toggleBreakpoint(11);
+        });
+        await disconnectDebugger(driver);
+        await (await new ActivityBar().getViewControl('Run and Debug')).closeView();
+        await killTerminal();
+        await new EditorView().closeAllEditors();
+    });
+
+    it('Update of a value with Camel debugger', async function () {
+        await executeCommand(CAMEL_RUN_DEBUG_ACTION_LABEL);
+        await waitUntilTerminalHasText(driver, TEST_ARRAY_RUN_DEBUG);
+        await driver.wait(async function () {
```

Comment:
Please verify timeout is finite in that case. I could not find it in Selenium documentation.

Reply:
:thinking: veeery nice catch, tested it is >100s, fixed